### PR TITLE
threadpool: wait for threads to start

### PR DESCRIPTION
**Pull request by bzoz**
Reopening https://github.com/libuv/libuv/pull/1637, since https://github.com/libuv/libuv/pull/1639 got reverted. From https://ci.nodejs.org/job/node-test-commit-windows-fanned/13917/ it looks like this does not break node.

Makes init_threads wait for all threads to start before continuing. Fixes some test flakiness on Windows.

CI: https://ci.nodejs.org/view/libuv/job/libuv-test-commit/610/

### CODE REVIEW

File: src/threadpool.c [hunk 1]
```diff
@@ -55,6 +58,11 @@ static void worker(void* arg) {
 
   (void) arg;
 
+  uv_mutex_lock(&threads_started_mutex);
+  ++threads_started;
+  uv_cond_signal(&threads_started_cond);
+  uv_mutex_unlock(&threads_started_mutex);
```
Comment (davisjam) [inline src/threadpool.c]: This is a hand-rolled semaphore.
The code would be more readable if a semaphore were used as described [here](https://github.com/libuv/libuv/pull/1639#issuecomment-348071153).

Is this actually [cheaper](https://github.com/libuv/libuv/pull/1639#issuecomment-348142522) than using a uv_sem_t?

**Comment by bnoordhuis** [comment]
I kind of agree with @davisjam that at this point we're just reimplementing a semaphore.  Let me see if I can code up a PR that uses a `uv_sem_t`.

**Comment by bnoordhuis** [comment]
https://github.com/libuv/libuv/pull/1662

**Comment by bzoz** [comment]
closing in favor of #1662

**Pull request closed without merging**